# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
# SPDX-License-Identifier: BSD-2-Clause
diff --git i/python/cudf/cudf/core/buffer/buffer.py w/python/cudf/cudf/core/buffer/buffer.py
index 71ab509348..e7e6ea9e5f 100644
--- i/python/cudf/cudf/core/buffer/buffer.py
+++ w/python/cudf/cudf/core/buffer/buffer.py
@@ -414,7 +414,7 @@ class Buffer(Serializable):
             "shape": (self.size,),
             "strides": None,
             "typestr": "|u1",
-            "version": 0,
+            "version": 3,
         }
 
     def serialize(self) -> tuple[dict, list]:
diff --git i/python/cudf/cudf/core/buffer/spillable_buffer.py w/python/cudf/cudf/core/buffer/spillable_buffer.py
index 52dcd00e2b..a5bc9a461d 100644
--- i/python/cudf/cudf/core/buffer/spillable_buffer.py
+++ w/python/cudf/cudf/core/buffer/spillable_buffer.py
@@ -337,7 +337,7 @@ class SpillableBufferOwner(BufferOwner):
             "shape": (self.size,),
             "strides": None,
             "typestr": "|u1",
-            "version": 0,
+            "version": 3,
         }
 
     def memoryview(
@@ -460,5 +460,5 @@ class SpillableBuffer(ExposureTrackedBuffer):
             "shape": (self.size,),
             "strides": None,
             "typestr": "|u1",
-            "version": 0,
+            "version": 3,
         }
diff --git i/python/cudf/cudf/core/column/column.py w/python/cudf/cudf/core/column/column.py
index 1562e4810c..a74a9fe23b 100644
--- i/python/cudf/cudf/core/column/column.py
+++ w/python/cudf/cudf/core/column/column.py
@@ -1971,7 +1971,7 @@ class ColumnBase(Serializable, BinaryOperand, Reducible):
             "strides": (self.dtype.itemsize,),
             "typestr": self.dtype.str,
             "data": (self.data_ptr, False),
-            "version": 1,
+            "version": 3,
         }
         if self.nullable and self.has_nulls():
             # Create a simple Python object that exposes the
